### PR TITLE
[feat]:[IDP-156]: Added RBAC configs for IDP Settings

### PR DESCRIPTION
**Pull request by sarthakkasat**
## Describe your changes
Added new Permission IDP_SETTINGS and related configs
Relevant docs [here](https://harness.atlassian.net/wiki/spaces/IDP/pages/21282816244/Access+Control+Tech+Spec+WIP)

##Jira Links
[IDP-156](https://harness.atlassian.net/browse/IDP-156)

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

[IDP-156]: https://harness.atlassian.net/browse/IDP-156?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/44338)

### CODE REVIEW

File: access-control/service/src/main/resources/io/harness/accesscontrol/roles/managed-roles.yml [hunk 1]
```diff
@@ -594,6 +594,7 @@ roles:
         - ccm_cloudAssetGovernanceEnforcement_edit
         - ccm_cloudAssetGovernanceEnforcement_view
         - ccm_cloudAssetGovernanceEnforcement_delete
+        - idp_idpsettings_manage
```
Comment (reetika04) [inline access-control/service/src/main/resources/io/harness/accesscontrol/roles/managed-roles.yml:597]: Please update the version

**Comment by sarthakkasat** [comment]
*[`access-control/service/src/main/resources/io/harness/accesscontrol/roles/managed-roles.yml` line 597 at r2](https://reviewable.io/reviews/harness/harness-core/44338#-NP22kQdtMfP1bXj2KZR-r2-597:-NPGUGGE8tgYOpRKuk1_:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/0b4dd824a5c3c9e7e0368154d754923db8adcdd9/access-control/service/src/main/resources/io/harness/accesscontrol/roles/managed-roles.yml#L597)):*
<details><summary><i>Previously, reetika04 (Reetika) wrote…</i></summary><blockquote>

Please update the version
</blockquote></details>

Done.

**Review by reetika04 — COMMENTED** [review]
> Reviewed 7 of 9 files at r1, 1 of 1 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44338)* status: 9 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained.

**Review by reetika04 — DISMISSED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44338)* status: 9 file(s) to review. 1 of 2 approvals obtained.

**Review by ankit-harness — DISMISSED** [review]
> Reviewed 7 of 9 files at r1, 1 of 1 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44338)* status: :shipit: complete! ✓

**Review by deepharness — APPROVED** [review]
> Reviewed 7 of 9 files at r1, 1 of 1 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44338)* status: :shipit: complete! ✓

**Review by ankit-harness — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44338#-:-NPH_C851eqTdEPk_U82:bnfp4nl)* status: 1 file(s) to review. 1 of 2 approvals obtained.

File: .gitleaksignore [hunk 2]
```diff
@@ -13,6 +13,7 @@ b54489244a91d374672d973f73f7481e0bd07f51:audit-event-streaming/service/src/main/
 b5b9c5604eea04a11f7f2da2343e7c83279b28e2:platform-service/config/config.yml:generic-api-key:164
 b5b9c5604eea04a11f7f2da2343e7c83279b28e2:platform-service/service/src/test/resources/test-config.yml:generic-api-key:152
 cba50989e673ec1c1999cbecfa687b0ff63f4999:360-cg-manager/config.yml:private-key:321
+bdbf341c917a79762b14831f763d6d1aecf337e7:idp-service/config/config.yml:generic-api-key:72
```
Comment (pranay-harness) [inline .gitleaksignore:16]: Approved. Dummy placeholder value which will be updated during runtime

**Review by pranay-harness — APPROVED** [review]

**Review by deepharness — APPROVED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44338)* status: :shipit: complete! ✓

**Pull request merged**